lib/spack/spack/cmd/logs.py

```diff
@@ -23,45 +25,36 @@ def setup_parser(subparser):
     arguments.add_common_arguments(subparser, ["spec"])
 
 
-def _dump_byte_stream_to_stdout(instream):
+def _dump_byte_stream_to_stdout(instream: io.BufferedIOBase) -> None:
+    # Reopen stdout in binary mode so we don't have to worry about encoding
     outstream = os.fdopen(sys.stdout.fileno(), "wb", closefd=False)
-
     shutil.copyfileobj(instream, outstream)
 
 
-def dump_build_log(package):
-    with open(package.log_path, "rb") as f:
-        _dump_byte_stream_to_stdout(f)
-
-
-def _logs(cmdline_spec, concrete_spec):
+def _logs(cmdline_spec: spack.spec.Spec, concrete_spec: spack.spec.Spec):
     if concrete_spec.installed:
         log_path = concrete_spec.package.install_log_path
     elif os.path.exists(concrete_spec.package.stage.path):
-        dump_build_log(concrete_spec.package)
-        return
+        # TODO: `spack logs` can currently not show the logs while a package is being built, as the
+        # combined log file is only written after the build is finished.
```

Comment:
@scheibelp do you agree that we should let Spack output both the phase and combined log while building, instead of assembling the build log post build (failure)?

Right now it's not possible to dump partial logs during the build, the file only exists after

Reply:
> do you agree that we should let Spack output both the phase and combined log while building,

Yes. That seems reasonable and I assumed it would look like https://github.com/spack/spack/pull/42202#discussion_r1468018994 (not that it has to, but my point is I think it would be relatively straightforward to get the `log_output` object to duplicate the logs).

> even if we don't have tail -f support.

My guess is if a user is excited enough to want to dump the logs during a build, they will want something that updates automatically (my general concern of not bundling `tail -f` is whether it would be obvious to a user whether they got a partial log, I think it's a matter of opinion though and wouldn't be opposed to a partial dump).

> Right now it's not possible to dump partial logs during the build,

Yes: for now `spack logs` isn't useful for dumping logs during builds, although the logs are also combined on a failed build, so it works for that.